src/sinks/amqp/config.rs

```diff
@@ -18,6 +18,10 @@ pub struct AmqpPropertiesConfig {
     /// Content-Encoding for the AMQP messages.
     #[configurable(derived)]
     pub(crate) content_encoding: Option<String>,
+
+    /// Expiration for AMQP messages (in milliseconds)
+    #[configurable(derived)]
+    pub(crate) expiration: Option<String>,
```

Comment:
Looking at the [docs](https://www.rabbitmq.com/docs/ttl#per-queue-message-ttl) this can be a `long-long-int` so we should make this a `u64` to prevent users misconfiguring this as a non integer value.

```suggestion
    pub(crate) expiration: Option<u64>,
```

Reply:
Agreed. We should also suffix it with the units per https://github.com/vectordotdev/vector/blob/master/docs/specs/configuration.md#option-naming so I think this would be `expiration_ms`.

Reply:
for whatever reason , Lapin coerces it from a String ( https://docs.rs/lapin/latest/lapin/protocol/basic/struct.AMQPProperties.html#method.with_expiration )